k8-operator/controllers/infisicalsecret_helper.go

```diff
@@ -225,12 +227,44 @@ func (r *InfisicalSecretReconciler) GetInfisicalServiceAccountCredentialsFromKub
 	return model.ServiceAccountDetails{AccessKey: string(accessKeyFromSecret), PrivateKey: string(privateKeyFromSecret), PublicKey: string(publicKeyFromSecret)}, nil
 }
 
+type TemplateSecret struct {
+	Value      string `json:"value"`
+	SecretPath string `json:"secretPath"`
+}
+
 func (r *InfisicalSecretReconciler) CreateInfisicalManagedKubeSecret(ctx context.Context, infisicalSecret v1alpha1.InfisicalSecret, secretsFromAPI []model.SingleEnvironmentVariable, ETag string) error {
 	plainProcessedSecrets := make(map[string][]byte)
 	secretType := infisicalSecret.Spec.ManagedSecretReference.SecretType
+	managedTemplateData := infisicalSecret.Spec.ManagedSecretReference.Template
+
+	if managedTemplateData == nil || managedTemplateData.IncludeAllSecrets {
+		for _, secret := range secretsFromAPI {
+			plainProcessedSecrets[secret.Key] = []byte(secret.Value) // plain process
+		}
+	}
+
+	if managedTemplateData != nil {
+		secretKeyValue := make(map[string]TemplateSecret)
+		for _, secret := range secretsFromAPI {
+			secretKeyValue[secret.Key] = TemplateSecret{
+				Value:      secret.Value,
+				SecretPath: secret.SecretPath,
+			}
+		}
+
+		for tmplKey, userTmpl := range managedTemplateData.Data {
+			tmpl, err := template.New("secret-templates").Parse(userTmpl)
```

Comment:
Can we make `secret-templates` a constant? I intend on adding templating support for the new PushSecret CRD in the future, and I think it makes sense to have the template names as a constant value

Reply:
You can put any name there. It doesn't matter at all. For push you can use another name that signifies it's a `PUSH` template.